examples/image_ocr.py

```diff
@@ -0,0 +1,433 @@
+        pylab.close()
+
+# Input Parameters
+img_h = 64
+img_w = 512
+nb_epoch = 50
+minibatch_size = 32
+total_words = 12480
+val_split = 0.2
+val_words = int(total_words * (val_split))
+
+# Network parameters
+conv_num_filters = 16
+filter_size = 3
+pool_size_1 = 4
+pool_size_2 = 2
+time_dense_size = 32
+rnn_size = 512
+time_steps = img_w / (pool_size_1 * pool_size_2)
+
+img_gen = TextImageGenerator(monogram_file="wordlist_mono_clean.txt",
+                             bigram_file="wordlist_bi_clean.txt",
+                             minibatch_size=32,
+                             img_w=img_w,
+                             img_h=img_h,
+                             downsample_width=img_w / (pool_size_1 * pool_size_2) - 2,
+                             num_words=total_words,
+                             val_split=total_words - val_words)
+
+act = 'relu'
+input_data = Input(name="the_input", shape=(1, img_h, img_w), dtype="float32")
+inner = Convolution2D(conv_num_filters, filter_size, filter_size, border_mode='same',
+                      activation=act, input_shape=(1, img_h, img_w), name="conv1")(input_data)
+inner = MaxPooling2D(pool_size=(pool_size_1, pool_size_1), name="max1")(inner)
+inner = Convolution2D(conv_num_filters, filter_size, filter_size, border_mode='same',
+                      activation=act, name="conv2")(inner)
+inner = MaxPooling2D(pool_size=(pool_size_2, pool_size_2), name="max2")(inner)
+
+conv_to_rnn_dims = ((img_h / (pool_size_1 * pool_size_2))*conv_num_filters,  img_w / (pool_size_1 * pool_size_2))
+inner = Reshape(target_shape=conv_to_rnn_dims, name="reshape")(inner)
+inner = Permute(dims=(2, 1), name="permute")(inner)
+
+# cuts down input size going into RNN:
+inner = TimeDistributed(Dense(time_dense_size, activation=act, name="dense1"))(inner)
+
+# Two layers of bidirecitonal GRUs
+# GRU seems to work as well, if not better than LSTM:
+gru_1 = GRU(rnn_size, return_sequences=True, name="gru1")(inner)
+gru_1b = GRU(rnn_size, return_sequences=True, go_backwards=True, name="gru1_b")(inner)
+gru1_merged = merge([gru_1, gru_1b], mode="sum")
+gru_2 = GRU(rnn_size, return_sequences=True, name="gru2")(gru1_merged)
+gru_2b = GRU(rnn_size, return_sequences=True, go_backwards=True)(gru1_merged)
+
+# transforms RNN output to character activations:
+inner = TimeDistributed(Dense(img_gen.get_output_size(), name="dense2"))(merge([gru_2, gru_2b], mode="concat"))
+sfm = Activation("softmax", name="softmax")(inner)
+Model(input=[input_data], output=[sfm]).summary()
+
+labels = Input(name="the_labels", shape=[img_gen.ABSOLUTE_MAX_STRING_LEN], dtype="float32")
+input_length = Input(name="input_length", shape=[1], dtype="int64")
+label_length = Input(name="label_length", shape=[1], dtype="int64")
+loss_out = merge([sfm, labels, input_length, label_length], mode=ctc_lambda_func, output_shape=(1,), name="ctc")
+
+if K._BACKEND == "tensorflow":  # maybe slight differences in CTC cause this?
```

Comment:
The Theano CTC implementation uses a different method for assuring numerical stability by normalizing before the final log.  The loss value for the Theano function ends up being lower, though it scales the same and still trains properly. So lowering TensorFlow's learning rate actually makes a big difference.  I think the better long term solution is to either A) fix the Theano CTC code so to match TensorFlows or B) wait for native CTC to be implemented in Theano, which they're working on.